hack/aad.sh

```diff
@@ -55,8 +56,19 @@ EOF
     fi
 
     cat <<EOF
-AZURE_CLIENT_ID=$AZURE_CLIENT_ID
-AZURE_CLIENT_SECRET=$AZURE_CLIENT_SECRET
+AZURE_AAD_CLIENT_ID=$AZURE_AAD_CLIENT_ID
+AZURE_AAD_CLIENT_SECRET=$AZURE_AAD_CLIENT_SECRET
+EOF
+    ;;
+
+app-update)
```

Comment:
I think that adding this is implying a new way of working with aad.sh, whereby create.sh/delete.sh no longer call aad.sh for you.  I'm comfortable with that, but I think we should try to find a cleaner interface which maintains ease of use.  Requests:
1. remove the calls to aad.sh from create.sh/delete.sh, and update README.md with usage examples.
2. make the app-update and app-delete interface for aad.sh more homogeneous and improve usage().  At the moment one function takes a hostname arg and looks up the ID; the other implicitly uses AZURE_AAD_CLIENT_ID, and I don't like that.  I think making the arguments explicit is clear but probably a pain to use.  Making them implicit is OK if we check the env var is set, and usage() and README.md are OK.
3. team e-mail to draw attention to the change when it merges.

Reply:
(btw if you hadn't added this in at the last minute to this PR, it would have merged by now :-/  A new PR would have been better.)

Reply:
> remove the calls to aad.sh from create.sh/delete.sh, and update README.md with usage examples.

I think it might stay as dummy `provider` for testing purpose. If we remove this at all we will need to populate those values in tests with dummies. This might be acceptable, dunno. 
Calling `app-update` from create.sh makes life little bit easier than creating multiple cluster with same AAD object. Removing it implies now I need to run `create.sh` and `aad.sh` and I need to know argumets (hostname and ID). 

> make the app-update and app-delete interface for aad.sh more homogeneous and improve usage(). At the moment one function takes a hostname arg and looks up the ID; the other implicitly uses AZURE_AAD_CLIENT_ID, and I don't like that. I think making the arguments explicit is clear but probably a pain to use. Making them implicit is OK if we check the env var is set, and usage() and README.md are OK.

I think adding checks should be enough. Considering `app-update` and `app-create` behaves differently.